core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -776,9 +777,11 @@ public Builder addSchema(final Path schemaPath) {
             if (null != schemaPath) {
                 try {
                     if (Files.isDirectory(schemaPath)) {
```

Comment:
Would try with resources work if the if was before the try?

Reply:
Yes it looks like it would. Everything in the `try` is inside an `if` and which can probably be merged with the other one as well as being taken out.